src/libasr/pass/intrinsic_function_registry.h

```diff
@@ -1160,6 +1162,53 @@ static inline void verify_args(const ASR::IntrinsicScalarFunction_t& x,
 
 } // namespace BinaryIntrinsicFunction
 
+
+namespace ObjectType {
+
+     static inline void verify_args(const ASR::IntrinsicScalarFunction_t& x, diag::Diagnostics& diagnostics) {
+        ASRUtils::require_impl(x.n_args != 1,
+            "ASR Verify: type() takes only 1 argument `object`",
+            x.base.base.loc, diagnostics);
+    }
+
+    static ASR::expr_t *eval_ObjectType(Allocator &al, const Location &loc,
+            ASR::ttype_t* t1, Vec<ASR::expr_t*>& args) {
+        std::string object_type = "<type '" + ASRUtils::type_to_str_python(t1) + "'>";
+        return StringConstant(object_type, character(object_type.length()));
+    }
+
+    static inline ASR::asr_t* create_ObjectType(Allocator& al, const Location& loc,
+            Vec<ASR::expr_t*>& args,
+            const std::function<void (const std::string &, const Location &)> err) {
+        if (args.size() > 1) {
+            err("type() takes only 1 argument `object`", loc);
+        }
```

Comment:
```suggestion
        if (args.size() == 1) {
            err("type() takes exactly 1 argument `object` for now", loc);
        }
```

Reply:
CPython has the `type()` overloaded. One of them takes just 1 argument `object`. While discussing with you earlier in the day, I was mistakenly referring to the other function which creates a new type - `type(name, dict, **kwargs)`.

This should remain unchanged I think.

Reply:
According to the implement in this PR, we use or handle or perform operation on the 1st argument, so I suggested the above change. Later, we can handle more arguments.

Reply:
@Thirumalai-Shaktivel did you mean to say `args.size() != 1`?

Reply:
Yup!